doc/man3/SSL_CTX_set1_cert_comp_preference.pod

```diff
@@ -0,0 +1,138 @@
+enabled in the library will be accepted in B<algs>, unknown algorithms in B<algs>
+are ignored. On an error, the preference order is left unmodified.
+
+The following compression algorithms (B<alg> arguments) may be used:
+
+=over 4
+
+=item TLSEXT_comp_cert_brotli
+
+=item TLSEXT_comp_cert_zlib
+
+=item TLSEXT_comp_cert_zstd
+
+=back
+
+The above is also the default preference order. If a preference order is not
+specified, then the default preference order is sent to the peer and the
+received peer's preference order will be used when compressing a certificate.
+Otherwise, the configured preference order is sent to the peer and is used
+when compressing the certificate.
+
+SSL_CTX_compress_certs() and SSL_compress_certs() are used to pre-compress all
+the configured certificates on an SSL_CTX (or SSL) with algorithm B<alg>. If
+B<alg> is 0, then the certificates are compressed with every configured
+algorithm in the library. Calling these functions on a client SSL_CTX/SSL will
+result in an error, as only server certificates may be pre-compressed.
+
+SSL_CTX_get1_compressed_cert() and SSL_get1_compressed_cert() are used to 
+get the pre-compressed certificate most recently set that may be stored for
+later use. Calling these functions on a client
+SSL_CTX/SSL will result in an error, as only server certificates may be
+pre-compressed. The B<data> and B<orig_len> arguments are required.
+
+The compressed certificate data may be passed to SSL_CTX_set1_compressed_cert()
+or SSL_set1_compressed_cert() to provide a pre-compressed version of the
+most recently set certificate. This pre-compressed certificate can only be used
+by a server.
+
+=head1 NOTES
+
+Only server certificates may be pre-compressed. Calling any of these functions
+(except SSL_CTX_set1_cert_comp_preferences()/SSL_set1_cert_comp_preferences())
+on a client SSL_CTX/SSL will return an error. Client certificates are compressed
+on-demand as unique context data from the server is compressed along with the
+certificate.
+
+For SSL_CTX_set1_cert_comp_preference() and SSL_set1_cert_comp_preference()
+the B<len> argument is the size of the B<algs> argument in bytes.
+
+The compressed certificate returned by SSL_CTX_get1_compressed_cert() and
+SSL_get1_compressed_cert() is the last certificate set on the SSL_CTX/SSL.
+The certificate memory is allocated by the function and the caller must
+free B<*data> via OPENSSL_free().
+
+The compressed certificate data set by SSL_CTX_set1_compressed_cert() and
+SSL_set1_compressed_cert() is copied into the SSL_CTX/SSL.
+
+Sending compressed certificates may be disabled on a connection via the
+SSL_OP_NO_TX_CERTIFICATE_COMPRESSION option. Receiving compressed certificates
+may be disabled on a connection via the SSL_OP_NO_RX_CERTIFICATE_COMPRESSION
+option.
+
+=head1 RETURN VALUES
+
+SSL_CTX_set1_cert_comp_preference(),
+SSL_set1_cert_comp_preference(),
+SSL_CTX_compress_certs(),
+SSL_compress_certs(),
```

Comment:
Is it an error in `SSL_CTX_compress_certs()` if no compression algorithms are available at all? We should document what happens.